mongodbatlas/fw_data_source_mongodbatlas_alert_configuration.go

```diff
@@ -321,124 +322,111 @@ func outputAlertConfiguration(alert *matlas.AlertConfiguration, outputType, reso
 	return map[string]cty.Value{
-		"metric_name": cty.StringVal(metric.MetricName),
-		"operator":    cty.StringVal(metric.Operator),
-		"threshold":   cty.NumberFloatVal(metric.Threshold),
-		"units":       cty.StringVal(metric.Units),
-		"mode":        cty.StringVal(metric.Mode),
+		"metric_name": cty.StringVal(*metric.MetricName),
+		"operator":    cty.StringVal(*metric.Operator),
+		"threshold":   cty.NumberFloatVal(*metric.Threshold),
+		"units":       cty.StringVal(*metric.Units),
+		"mode":        cty.StringVal(*metric.Mode),
 	}
 }
 
-func convertThresholdToCtyValues(threshold matlas.Threshold) map[string]cty.Value {
+func convertThresholdToCtyValues(threshold *admin.GreaterThanRawThreshold) map[string]cty.Value {
 	return map[string]cty.Value{
-		"operator":  cty.StringVal(threshold.Operator),
-		"units":     cty.StringVal(threshold.Units),
-		"threshold": cty.NumberFloatVal(threshold.Threshold),
+		"operator":  cty.StringVal(*threshold.Operator),
+		"units":     cty.StringVal(*threshold.Units),
+		"threshold": cty.NumberFloatVal(float64(*threshold.Threshold)), // int in new SDK but keeping float64 for backward compatibility
 	}
 }
 
-func convertNotificationToCtyValues(notification *matlas.Notification) map[string]cty.Value {
+func convertNotificationToCtyValues(notification *admin.AlertsNotificationRootForGroup) map[string]cty.Value {
 	values := map[string]cty.Value{}
 
-	if notification.ChannelName != "" {
-		values["channel_name"] = cty.StringVal(notification.ChannelName)
-	}
-
-	if notification.DatadogRegion != "" {
-		values["datadog_region"] = cty.StringVal(notification.DatadogRegion)
-	}
-
-	if notification.EmailAddress != "" {
-		values["email_address"] = cty.StringVal(notification.EmailAddress)
+	if conversion.IsStringPresent(notification.ChannelName) {
+		values["channel_name"] = cty.StringVal(*notification.ChannelName)
 	}
 
-	if notification.FlowName != "" {
-		values["flow_name"] = cty.StringVal(notification.FlowName)
+	if conversion.IsStringPresent(notification.DatadogRegion) {
+		values["datadog_region"] = cty.StringVal(*notification.DatadogRegion)
 	}
 
-	if notification.IntervalMin > 0 {
-		values["interval_min"] = cty.NumberIntVal(int64(notification.IntervalMin))
+	if conversion.IsStringPresent(notification.EmailAddress) {
+		values["email_address"] = cty.StringVal(*notification.EmailAddress)
 	}
 
-	if notification.MobileNumber != "" {
-		values["mobile_number"] = cty.StringVal(notification.MobileNumber)
+	if notification.IntervalMin != nil && *notification.IntervalMin > 0 {
+		values["interval_min"] = cty.NumberIntVal(int64(*notification.IntervalMin))
 	}
 
-	if notification.OpsGenieRegion != "" {
-		values["ops_genie_region"] = cty.StringVal(notification.OpsGenieRegion)
+	if conversion.IsStringPresent(notification.MobileNumber) {
+		values["mobile_number"] = cty.StringVal(*notification.MobileNumber)
 	}
 
-	if notification.OrgName != "" {
```

Comment:
same here for OrgName, I am assuming the new SDK does not require it?

Reply:
correct, they are for a notification type that doesn't exist any more